### ISSUE TITLE
CA1801: Parameters used inside local functions are incorrectly flagged as unused

### ISSUE THREAD

**Issue by srivatsn** [dotnet/roslyn-analyzers#1224]
#### Analyzer package

2.3.0-beta1

#### Analyzer

CA1801: Review unused parameters

#### Repro steps

```csharp
       internal static string EscapeString(string unescaped, char[] toEscape)
        {
            if (string.IsNullOrWhiteSpace(unescaped)) return unescaped;

            bool ShouldEscape(char c)
            {
                foreach (var escapeChar in toEscape)
                {
                    if (escapeChar == c) return true;
                }
                return false;
            }
        ...
```

#### Expected behavior
No warnings since toEscape here is being used in a local function
#### Actual behavior
toEscape is flagged as an unused param.

**Comment by mavasani** [comment]
I have a Roslyn PR out to fix this.

**Comment by jinujoseph** [comment]
This is tracked here https://github.com/dotnet/roslyn/issues/19902
